bokeh/models/tiles.py

```diff
@@ -26,11 +26,11 @@ class TileSource(PlotObject):
     the maximum zoom level for the tile layer. This is the most "zoomed-in" level.
     """)
 
-    x_origin_offset = Float(default=20037508.34, help="""
+    x_origin_offset = Float(default=None, help="""
```

Comment:
So one potential issue is that the object hierarchy:

Python:
TileSource -> (WMTSTileSource, TMSTileSource, QUADKEYTileSource, BBoxTileSource)

CoffeeScript
TileSource -> `MercatorTileSource` -> (WMTSTileSource, TMSTileSource, QUADKEYTileSource, BBoxTileSource)

The changed defaults arise at the `MercatorTileSource` level.  This seems like it would be good to refactor, but would probably be better on a separate PR.